### PR TITLE
Shooter Subsystem Done?

### PR DESCRIPTION
**Pull request by foxinabox384**
Hopefully achieved all tasks stated on issue #9. But it's done... so...

### CODE REVIEW

File: ShooterSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package frc.robot.subsystems.shooter;
```
Comment (ebay-kid) [inline ShooterSubsystem.java]: Delete this file, you already have the correct one in the correct folder with newer code

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    private final RelativeEncoder leaderEncoder;
+    private final RelativeEncoder pivotEncoder;
+    private final RelativeEncoder flapEncoder;
+    //make the limit switch variables
+    private final DigitalInput pivotLimitSwitch;
+    private final DigitalInput flapLimitSwitch;
+    //PID stuff
+    private static SparkPIDController pivotPidController;
+    private static SparkPIDController flapPidController;
+    private static SparkPIDController leaderPidController;
```
Comment (ebay-kid) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java]: these should just be final, not static

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        flapPidController.setI(0);
+        flapPidController.setD(0);
+        flapPidController.setFF(0);
+
+        leaderPidController.setP(0);
+        leaderPidController.setI(0);
+        leaderPidController.setD(0);
+        leaderPidController.setFF(0);
+    
+        }
```
Comment (ebay-kid) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java]: fix this indent

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    private static SparkPIDController pivotPidController;
+    private static SparkPIDController flapPidController;
+    private static SparkPIDController leaderPidController;
+    
+
+
+    public ShooterSubsystem(){
+        // intialize motors
+        followerMotor = new CANSparkMax(PortConstants.SHOOTER_FOLLOWER_MOTOR_ID, MotorType.kBrushless);
+        leaderMotor = new CANSparkMax(PortConstants.SHOOTER_LEADER_MOTOR_ID, MotorType.kBrushless);
```
Comment (ebay-kid) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java]: you should name the constant in the Constants file TOP/BOTTOM so it's a bit easier to understand when wiring. the names for your motor objects are good though

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+        leaderPidController.setP(0);
+        leaderPidController.setI(0);
+        leaderPidController.setD(0);
+        leaderPidController.setFF(0);
+    
+        }
+
+        // Setting the speed of the motors
+        public void setLeaderMotorSpeed(double RPM){
```
Comment (ebay-kid) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java]: name RPM lowercase (`rpm`)

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+        // Setting 
+        public void setPivotPosition(double position){
+            pivotEncoder.setPosition(position);
+        }
+        public void setFlapPosition(double position){
+            flapEncoder.setPosition(position);
+        }
+        public void setPivotPid(double degree){
+            pivotPidController.setP(degree);
```
Comment (ebay-kid) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java]: when setting PIDs, use `pidController.setReference(target, ControlType);`

File: src/main/java/frc/robot/Constants.java [hunk 7]
```diff
@@ -30,6 +30,16 @@ public static class ControllerConstants {
      * 12 (COD)
      */
     public static class PortConstants {
+
+        public static final int SHOOTER_FOLLOWER_MOTOR_ID           = 42; //THE MEANING OF LIFE :) FAKE ID, CHANGE LATER! 
+        public static final int SHOOTER_LEADER_MOTOR_ID             = 64; //THESE ARE ALL FAKE IDs
+        public static final int SHOOTER_FLAP_MOTOR_ID               = 23; 
+        public static final int SHOOTER_PIVOT_MOTOR_ID              = 54;
+        public static final int SHOOTER_FOLLOWER_ENCODER_ID         = 63; 
+        public static final int SHOOTER_LEADER_ENCODER_ID           = 34;
```
Comment (ebay-kid) [inline src/main/java/frc/robot/Constants.java]: no need for an ID for the encoder in this situation - we are using the integrated ones

**Review by ebay-kid — COMMENTED** [review]
Almost done, small changes

File: src/main/java/frc/robot/Robot.java [hunk 8]
```diff
@@ -52,6 +52,13 @@ public void robotPeriodic() {
         this.robotContainer.onTeleopPeriodic();
     }
 
+
+
+
+
+
+
+
```
Comment (yavko) [inline src/main/java/frc/robot/Robot.java]: whats all this whitespace for lol

File: src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java [hunk 9]
```diff
@@ -63,76 +63,75 @@ public ShooterSubsystem(){
...
         public void setFlapMotorSpeed(double degrees){
             flapMotor.set(degrees);
-        }
+    }
 
         // Getting
         public double getPivotPosition(){
             return pivotEncoder.getPosition();
-        }
+    }
```
Comment (yavko) [inline src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java:87]: Why break the consistent indentation

**Review by yavko — COMMENTED** [review]
I just wanted to mention some of the formatting issues I saw when taking a peak

**Comment by ebay-kid** [comment]
You're still missing a couple of the previous comments (specifically the one about setting PIDs). Also your whitespace and indents are off by a little bit like yavko said.

**Pull request merged**